src/vhpi/vhpi-model.c

```diff
@@ -1454,8 +1758,36 @@ void vhpi_build_design_model(tree_t top, rt_model_t *m)
 
    model = m;
 
+   tool = new_object(sizeof(c_tool), vhpiToolK);
+
+   c_designUnit *unit;
+   tree_t s = tree_primary(top);
```

Comment:
```suggestion
   tree_t hier = tree_decl(b0, 0);
   assert(tree_kind(hier) == T_HIER);

   tree_t s = tree_ref(hier);
```

That would then work with instances inside the hierarchy. Sorry it's not very obvious how to do that. Not sure if you were aware but you can set `NVC_ELAB_VERBOSE=1` to print the elaborated design and then the `T_HIER` nodes are printed as `-- Instance foo`.